src/providers/swift.rs

```diff
@@ -0,0 +1,147 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+use anyhow::Result;
+use indoc::formatdoc;
+
+static DEFAULT_SWIFT_VERSION: &str = "5.6.1";
+
+pub struct SwiftProvider {}
+
+impl Provider for SwiftProvider {
+    fn name(&self) -> &str {
+        "swift"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("Package.swift")
+            || (app.includes_file("Package.swift") && app.includes_file("Package.resolved")))
+    }
+
+    fn setup(&self, _app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        let pkgs = vec![Pkg::new("python38"), Pkg::new("wget")];
+
+        Ok(Some(SetupPhase::new(pkgs)))
+    }
+
+    fn install(&self, app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
+        let version = SwiftProvider::get_swift_version(&app)?;
+        let (download_url, name) = SwiftProvider::make_download_url(version);
+
+        // https://forums.swift.org/t/which-clang-package-should-we-install/20542/14
+        let install_cmd = formatdoc! {"
+        sudo apt-get update && \
+        sudo apt-get install -y build-essential clang libsqlite3-0 libncurses6 libcurl4 libxml2 libatomic1 libedit2 libsqlite3-0 libcurl4 libxml2 libbsd0 libc6-dev && \
```

Comment:
I am very hesitant to add a provider that installs random packages from apt-get like this. What is wrong with the Swift from nixpkgs? It is available on unstable https://search.nixos.org/packages?channel=unstable&show=swift&from=0&size=50&sort=relevance&type=packages&query=swift

Reply:
It's broken! [Uploading u2{95c13ee} (3557aab1-84a7-4f02-b35b-aae311858d1d id=0x7f0a02b4) …]()

Reply:
![unknown-3.png](https://user-images.githubusercontent.com/75403863/172268083-c648ce7f-04cb-430a-a24e-5279d78a2195.png)

Reply:
old versions also don't work

Reply:
I think instead of installing Swift from source, which slightly goes against the philosophy of nixpacks, we should instead fix the upstream Swift nix expression or create an overlay (similar to the [Rust overlay](https://github.com/oxalica/rust-overlay)) that has working versions of Swift. Swift is not necessary to reach parity with the buildpacks and I would like to avoid making this a standard way of installing language dependencies. However, I do really appreciate this PR and the effort that was put into it.